packages/nvidia_nat_a365/tests/telemetry/test_registration.py

```
@@ -0,0 +1,68 @@
+
+
+def test_a365_telemetry_exporter_config_accepts_auth_ref():
+    """Test that A365TelemetryExporter accepts AuthenticationRef for token_resolver."""
+    from nat.data_models.component_ref import AuthenticationRef
+
+    # Pydantic will coerce strings to AuthenticationRef automatically
+    # So both string and AuthenticationRef should work
+    config1 = A365TelemetryExporter(
+        agent_id="test-agent",
+        tenant_id="test-tenant",
+        token_resolver="test_auth",  # String is coerced to AuthenticationRef
```

Ruff (0.15.12)

[error] Possible hardcoded password assigned to argument: "token_resolver"

(S106)